src/Pate/Monad.hs

```diff
@@ -423,8 +429,27 @@ withAssumptionSet asm f = withSym $ \sym -> withSolverProcess $ \sp -> do
   CMR.local (\env -> env { envCurrentFrame = Some (asm <> curAsm) }) $ do
     IO.withRunInIO $ \inIO -> WPO.inNewFrame sp $ do
       W4.assume (WPO.solverConn sp) p
+      inIO $ validateAssumptions curAsm asm
       inIO f
 
+validateAssumptions ::
+  forall sym arch v. 
+  HasCallStack =>
+  AssumptionSet sym v ->
+  AssumptionSet sym v ->
+  EquivM sym arch ()
+validateAssumptions oldAsm newAsm = withSym $ \sym -> withSolverProcess $ \sp -> IO.withRunInIO $ \inIO -> do
+  let
+    simp :: forall tp. W4.SymExpr sym tp -> IO (W4.SymExpr sym tp)
+    simp e = resolveConcreteLookups sym (\e1 e2 -> W4.asConstantPred <$> liftIO (W4.isEq sym e1 e2)) e  >>= simplifyBVOps sym >>= expandMuxEquality sym
+  
+  oldAsm' <- liftIO $ PEM.mapExpr sym simp oldAsm
+  newAsm' <- liftIO $ PEM.mapExpr sym simp newAsm
```

Comment:
Presumably these simplifications are solely for readability?

Reply:
Yes. I'll put a comment in, but in general it would be good to have this kind of presentation-specific simplification logic exist in one place.

Reply:
https://github.com/GaloisInc/pate/issues/315